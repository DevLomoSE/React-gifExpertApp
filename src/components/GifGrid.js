import React, { useEffect, useState } from "react";
import GifGridItem from "./GifGridItem";

/**
 * Component to retrieve images about the category.
 *
 * @param category search criteria
 */
const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async () => {
    const url = `${process.env.REACT_APP_GIPHY_GIFS_LINK}/search?q=${category}&limit=10&api_key=${process.env.REACT_APP_GIPHY_API_KEY}`;

    const resp = await fetch(url);
    const { data } = await resp.json();

    const gifs = data.map((img) => {
      return {
        id: img.id,
        title: img.title,
        img: img.images?.downsized_medium.url,
      };
    });

    setImages(gifs);
  };

  return (
    <>
      <h3>{category}</h3>
      {images.map( img => {
        return (
            <>
                <GifGridItem 
                    key={img.id}
                    {...img}
                />  
            </>
        );
      })}
    </>
  );
};

export default GifGrid;
