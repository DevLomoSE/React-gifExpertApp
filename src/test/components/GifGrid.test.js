import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import "@testing-library/jest-dom";

import "../../setupTests";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe("Testing for <GifGrid />", () => {

  test("should render", () => {

    useFetchGifs.mockReturnValue({
        data: [],
        loading: true  
    });

    const wrapper = shallow(<GifGrid />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show items while calling useFetchGifs', () => {

    const gifs = [{
        id: 'lomo',
        url: '127.0.0.1/lomo',
        title: 'lomo'
    }]

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: true,
    });
    
    const wrapper = shallow(<GifGrid />);
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });

  test('should not to be GifGridItem', () => {

      useFetchGifs.mockReturnValue({
        data: [],
        loading: true,
      });

      const wrapper = shallow(<GifGrid />);
      expect(wrapper).toMatchSnapshot();

      expect(wrapper.find("GifGridItem").length).toBe(0);
  });

});
