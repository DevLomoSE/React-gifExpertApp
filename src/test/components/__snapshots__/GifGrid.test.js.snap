// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing for <GifGrid /> should not to be GifGridItem 1`] = `
<Fragment>
  <h3 />
  <p
    className="animate__animated animate__pulse"
  >
    " Cargando..."
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Testing for <GifGrid /> should render 1`] = `
<Fragment>
  <h3 />
  <p
    className="animate__animated animate__pulse"
  >
    " Cargando..."
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Testing for <GifGrid /> should show items while calling useFetchGifs 1`] = `
<Fragment>
  <h3 />
  <p
    className="animate__animated animate__pulse"
  >
    " Cargando..."
  </p>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="lomo"
      key="lomo"
      title="lomo"
      url="127.0.0.1/lomo"
    />
  </div>
</Fragment>
`;
